### PR TITLE
Add Flush

### PR DESCRIPTION
**Pull request by ogaca-dd**
This PR introduces the `Flush` method in order to flush without disposing.

It should fix https://github.com/DataDog/dogstatsd-csharp-client/issues/141

### CODE REVIEW

File: src/StatsdClient/DogStatsdService.cs [hunk 1]
```diff
@@ -242,6 +242,15 @@ public void ServiceCheck(string name, Status status, int? timestamp = null, stri
             _metricsSender?.SendServiceCheck(name, (int)status, timestamp, hostname, tags, message);
         }
 
+        /// <summary>
+        /// Flushes all metrics.
+        /// </summary>
+        public void Flush()
+        {
+            _statsdData?.Flush();
+            _statsdData = null;
```
Comment (danny-zegel-zocdoc) [inline src/StatsdClient/DogStatsdService.cs:251]: you dont need to create a new `_statsdData`?

Comment (ogaca-dd) [reply src/StatsdClient/DogStatsdService.cs]: Fixed.

File: src/StatsdClient/Worker/AsynchronousWorker.cs [hunk 2]
```diff
@@ -55,29 +58,41 @@ public bool TryEnqueue(T value)
...
 
-            _terminate = true;
-            try
+            _requestFlush = true;
+            _flushEvent.WaitOne(maxWaitDurationInFlush);
+        }
+
+        public void Dispose()
+        {
+            if (!_terminate)
```
Comment (gh123man) [inline src/StatsdClient/Worker/AsynchronousWorker.cs:76]: NIT: could flip this to reduce nesting:
`if (_terminate) { return } `

**Review by truthbk — APPROVED** [review]

File: src/StatsdClient/Worker/AsynchronousWorker.cs (hunk 2 above)
Comment (ogaca-dd) [reply src/StatsdClient/Worker/AsynchronousWorker.cs:76]: I tend to use this pattern where there are several `return` in the function.

**Comment by danny-zegel-zocdoc** [comment]
any idea when a new version of https://www.nuget.org/packages/DogStatsD-CSharp-Client will be released with this?

**Comment by ogaca-dd** [comment]
[DogStatsD-CSharp-Client 6.0.0](https://www.nuget.org/packages/DogStatsD-CSharp-Client/6.0.0) has been released.
See changelog: https://github.com/DataDog/dogstatsd-csharp-client/releases/tag/6.0.0

**Pull request merged**